CoreAttributeEditorView/src/au/gov/asd/tac/constellation/views/attributeeditor/editors/IconEditorFactory.java

```diff
@@ -246,62 +249,92 @@ private void addNode(TreeItem<IconNode> item, IconNode node) {
-                }
+                final FileChooserBuilder iconEditorFileChooser = getIconEditorFileChooser()
+                        .setFilesOnly(true)
+                        .setFileFilter(new FileFilter() {
+                            @Override
+                            public boolean accept(final File file) {
+                                final String name = file.getName();
+                                return (file.isFile() && StringUtils.endsWithIgnoreCase(name, FileExtensionConstants.PNG)) || file.isDirectory();
+                            }
+
+                            @Override
+                            public String getDescription() {
+                                return "Image Files (" + FileExtensionConstants.PNG + ")";
+                            }
+                        });
+
+                FileChooser.openMultiDialog(iconEditorFileChooser).thenAccept(optionalFiles -> optionalFiles.ifPresent(files -> addIcons(files)));
             });
+
             addDirButton.setOnAction(event -> {
-                final DirectoryChooser addFolderChooser = new DirectoryChooser();
-                addFolderChooser.setTitle("Add new Icons");
-                final File selectedFolder = addFolderChooser.showDialog(null);
-                if (selectedFolder != null) {
-                    final List<File> selectedFiles = pngWalk(selectedFolder);
-                    if (!selectedFiles.isEmpty()) {
-                        String lastIconFile = null;
-                        for (final File file : selectedFiles) {
-                            final String iconName = file.getName().substring(0, file.getName().length() - 4);
-                            final ConstellationIcon customIcon = new ConstellationIcon.Builder(iconName, new FileIconData(file)).build();
-                            if (IconManager.addIcon(customIcon)) {
-                                lastIconFile = file.getName().substring(0, file.getName().length() - 4);
-                            }
-                        }
-                        if (lastIconFile != null) {
-                            reloadUserDefinedIcons(lastIconFile);
-                        }
-                    }
-                }
+                final FileChooserBuilder iconEditorFileChooser = getIconEditorFileChooser().setDirectoriesOnly(true);
+                FileChooser.openOpenDialog(iconEditorFileChooser).thenAccept(optionalFolder -> optionalFolder.ifPresent(folder -> addIcons(pngWalk(folder))));
             });
+
             removeButton.setOnAction(event -> {
                 final boolean iconRemoved = IconManager.removeIcon(listView.getSelectionModel().getSelectedItem());
                 if (iconRemoved) {
                     reloadUserDefinedIcons("");
                 }
             });
-            addRemoveBox.getChildren().addAll(addFilesButton, addDirButton, removeButton);
+
+            // Added for testing
+            clearAllButton.setOnAction(event -> {
```

Comment:
if this has only been added for testing, is there a way you can include the code in this button in the unit test class without incorporating it here? I don't think it should be included as a feature if it serves no purpose outside of it